test/integration/testpaths/testpath_delta_snapshots.py

```diff
@@ -0,0 +1,516 @@
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            for snapshot in self.snapshots_created:
+                snapshot.delete(self.apiclient)
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @classmethod
+    def RestartServer(cls):
+        """Restart management server"""
+
+        sshClient = SshClient(
+            cls.mgtSvrDetails["mgtSvrIp"],
+            22,
+            cls.mgtSvrDetails["user"],
+            cls.mgtSvrDetails["passwd"]
+        )
+        command = "service cloudstack-management restart"
+        sshClient.execute(command)
+
+        return
+
+    @attr(tags=["advanced", "basic"], required_hardware="true")
+    def test_02_delta_snapshots(self):
+        """ Delta Snapshots
+            1. Create file on ROOT disk of deployed VM.
+            2. Create Snapshot of ROOT disk.
+            3. Verify secondary storage count.
+            4. Check integrity of Full Snapshot.
+            5. Delete delta snaphshot and check integrity of\
+               remaining snapshots.
+            6. Delete full snapshot and verify it is deleted from\
+               secondary storage.
+        """
+
+        checksum_created = []
+        full_snapshot_count = 0
+        delta_snapshot_count = 0
+
+        # Mulitply delta max value by 2 to set loop counter
+        # to create 2 Snapshot chains
+        snapshot_loop_count = int(self.delta_max) * 2
+
+        # Get ROOT Volume
+        root_volumes_list = list_volumes(
+            self.apiclient,
+            virtualmachineid=self.vm.id,
+            type='ROOT',
+            listall=True
+        )
+
+        status = validateList(root_volumes_list)
+        self.assertEqual(
+            status[0],
+            FAIL,
+            "Check listVolumes response for ROOT Disk")
+
+        root_volume = root_volumes_list[0]
+
+        # Get Secondary Storage Value from Database
+        qryresult_before_snapshot = self.dbclient.execute(
+            " select id, account_name, secondaryStorageTotal\
+                    from account_view where account_name = '%s';" %
+            self.account.name)
+
+        self.assertNotEqual(
+            len(qryresult_before_snapshot),
+            0,
+            "Check sql query to return SecondaryStorageTotal of account")
+
+        storage_qry_result_old = qryresult_before_snapshot[0]
+        secondary_storage_old = storage_qry_result_old[2]
+
+        # Create Snapshots
+
+        for i in range(snapshot_loop_count):
+
+            # Step 1
+            checksum_root = createChecksum(
+                self.testdata,
+                self.vm,
+                root_volume,
+                "rootdiskdevice")
+
+            time.sleep(30)
```

Comment:
Checksum is appended in a list immediately after creating it, and that checksum is used later to check integrity of snapshots.
Since it takes a few seconds to create checksum, sleep is added.
If sleep is not added, the integrity fails.